src/app/qgisapp.cpp

```diff
@@ -11395,7 +11395,7 @@ QgsVectorLayer *QgisApp::addVectorLayerPrivate( const QString &vectorLayerPath,
   {
     if ( guiWarning )
     {
-      QString message = layer->dataProvider()->error().message( QgsErrorMessage::Text );
+      QString message = layer->dataProvider() ? layer->dataProvider()->error().message( QgsErrorMessage::Text ) : tr( "Invalid provider" );
```

Comment:
This looks like a bug fix which should be applied to 3.10 also?

Reply:
Maybe, but unlikely to happen in real world situations. I noticed this in an intermediate stage where I tried to add a layer for a provider that doesn't exist.